### PR TITLE
[SPARK-7225][SQL] CombineLimits optimizer does not work

### PR DESCRIPTION
**Pull request by pzzs**
SQL 

```
select key from (select key from src limit 100) t2 limit 10
```

Optimized Logical Plan before modifying

```
== Optimized Logical Plan ==
Limit 10
Limit 100
Project key#3
MetastoreRelation default, src, None
```

Optimized Logical Plan after modifying

```
== Optimized Logical Plan ==
Limit 10
 Project [key#1]
  MetastoreRelation default, src, None
```

### CODE REVIEW

**Comment by rxin** [comment]
Thanks - what is the change exactly that made it work? Can we have a unit test for this?

**Comment by DoingDone9** [comment]
@rxin  i have added a test fot this.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala [hunk 1]
```diff
@@ -36,6 +36,13 @@ object DefaultOptimizer extends Optimizer {
...
       EliminateSubQueries) ::
+    Batch("Filter Pushdown", FixedPoint(100),
+      UnionPushdown,
+      CombineFilters,
+      PushPredicateThroughProject,
+      PushPredicateThroughJoin,
+      PushPredicateThroughGenerate,
+      ColumnPruning) ::
     Batch("Combine Limits", FixedPoint(100),
       CombineLimits) ::
```
Comment (marmbrus) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala]: Instead, can we just move `CombineLimits` down into the `Filter Pushdown` batch?

Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala]: (and rename "Filter Pushdown" into "Operator Reordering")

**Comment by DoingDone9** [comment]
git it   @marmbrus  @rxin

**Comment by rxin** [comment]
Jenkins, retest this please.

**Comment by rxin** [comment]
Thanks. Merging in master.

**Pull request closed without merging**